### DISCUSSION TITLE
Proper access to checkout block parts

### DISCUSSION THREAD

**Discussion by alexhopkin** [woocommerce/woocommerce#45337] [Checkout Flow]
It would be great if we were able to freely move the checkout block parts around within the patterns (for example moving the email address capture below the billing information), alongside allowing us to move specific parts to different columns (one example would be moving the order notes checkbox to the right-hand column). There may be a way to do this but it doesn't seem as simple as being able to use the blocks anywhere, and I can only lock/unlock certain blocks.

**Comment by senadir** [comment]
Hey! It's not yet possible as we believe the current structure serves best when it comes to B2C ecommerce websites, but adding the ability to move blocks is in our radar.

One question is that what justification do you see to move them? As we're more likely to accelerate this effort if we have good feedback. The email is very common to be captured first in all ecommerce platforms, and the order note is an actionable, fillable part of Checkout, so it's in the main column, vs having summary items, in the summary on the right.

**Reply by alexhopkin** [reply]
Hi, thanks for the reply. We build custom themes, and whilst we're able to approach building out different layouts by using filters, WC fragments, or custom CSS to meet the needs/requests of our clients, it would be easier to be able to shift the blocks around directly rather than going through workarounds.

This particular thought came from building out a site that solely sells virtual products, but the client wanted the capacity to move the columns around to fill out the white space left on the right hand column. I do agree that the form follows a typical data capture flow (and providing a standardised layout by default is important for this), however it could be argued that the order notes checkbox may be missed, so the capacity to move this to a prominent area above the fold may help customers to find and input their notes more easily. 

I've noticed that some of the blocks are locked with the option to unlock and move these around within their parent, but it feels a little inconsistent as to which ones have this functionality and which don't.
